### ISSUE TITLE
PyDMEmbeddedDisplay does not handle relative paths from Qt Designer

### ISSUE THREAD

**Issue by ivany4** [slaclab/pydm#508]
When PyDMEmbeddedDisplay is rendered in Qt Designer, it will fail to locate the file that is relative to the *.ui file opened in Qt Designer. It will also disregard `PYDM_DISPLAYS_PATH` variable.

**Expected**
Qt Designer path resolution should be identical to the runtime.

**Steps to Reproduce**
Create a `main_file.ui` and add PyDMEmbeddedDisplay widget that refers to a filename, e.g. `embedded_file.ui` (can be in the same directory as `main_file.ui`, as long as `designer` is called from the different directory).
```bash
designer some/relative/path/main_file.ui
```
Check the output
```
FileNotFoundError: [Errno 2] No such file or directory: 'embedded_file.ui'
Exception while opening embedded display file.
```

**Possible Solution**
`PYDM_DISPLAYS_PATH` case can be solved by copying logic from `PyDMApplication.open_relative()`, but I haven't yet figured out how to determine the relative path to the `main_file.ui`, as in runtime it's done via `PyDMApplication.directory_stack`, which is not available in Qt Designer.

**Additional context**
Introduced in v1.7, by [this commit](https://github.com/slaclab/pydm/commit/9885475560907d36e66a29ffcab5fd9fb3e79d69).

**Comment by mattgibbs** [comment]
Hi @ivany4, your 'Possible Solution' comments get to the root of the issue: In order to handle relative file paths in nested embedded displays, PyDM needs some way to keep track of the directory in which each embedded display file lives.  Historically, we've relied on PyDMApplication to do this, mostly because it was once responsible for opening all files, so it was convenient to handle everything there.

I think it might be possible to do this without PyDMApplication by traversing up the embedded display's parent tree, and looking for other embedded displays and their file paths.  We'd have to be careful to also keep track of all macro variables too, so that they can be properly cascaded.

The issue of `PYDM_DISPLAYS_PATH` is a little easier, I think we can use `pydm.utilities.find_display_in_path()` inside `PyDMEmbeddedDisplay.open_file()`.

**Comment by hhslepicka** [comment]
Closed via #590
